hspec-discover/src/Test/Hspec/Discover/Run.hs

```diff
@@ -62,24 +62,20 @@ mkSpecModule src conf nodes =
   ( "{-# LINE 1 " . shows src . " #-}\n"
   . showString "{-# LANGUAGE NoImplicitPrelude #-}\n"
   . showString "{-# OPTIONS_GHC -fno-warn-warnings-deprecations #-}\n"
-  . showString ("module " ++ moduleName src conf ++" where\n")
+  . showString ("module " ++ moduleName ++ exports ++ " where\n")
   . importList nodes
   . showString "import Test.Hspec.Discover\n"
   . maybe driver driverWithFormatter (configFormatter conf)
   . showString "spec :: Spec\n"
   . showString "spec = "
   . formatSpecs nodes
   ) "\n"
-  where
-    driver =
-        case configNoMain conf of
-          False ->
-              showString "main :: IO ()\n"
-            . showString "main = hspec spec\n"
-          True -> ""
-
-moduleName :: FilePath -> Config -> String
-moduleName src conf = fromMaybe (if configNoMain conf then pathToModule src else "Main") (configModuleName conf)
+ where
+  ifNoMain x y = if configNoMain conf then x else y
+  moduleName = fromMaybe (ifNoMain (pathToModule src) "Main")
+                         (configModuleName conf)
```

Comment:
Please don’t align code. Why you ask? If somebody later renames `moduleName` (or adds a parameter) then the alignment will break. This is bad for two reasons:

- It discourages refactoring by making it more expensive
- It produces a larger diff, making rebases harder / conflicts more likely

It’s not a big issue here, but it’s best to not get into the habit of aligning code in the first place. At the point you align record fields, etc, it’s more of an issue.

The short story is, aligned code looks nice when you write it, but if you “design your code for change”, it’s not an effective approach.

Reply:
The second line and indentation/alignment arose because I was trying to avoid long code lines. I've found a way to acheive both objectives.